### PR TITLE
refactor: simplify automatic migration code by using client keeper functions

### PR DESCRIPTION
**Pull request by colin-axner**
## Description

Skipping changelog for code cleanup of unreleased changes

closes: #2845 

### Commit Message / Changelog Entry

```bash
type: commit message
```

see the [guidelines](https://github.com/cosmos/ibc-go/blob/main/CONTRIBUTING.md#commit-messages) for commit messages. (view raw markdown for examples)

---

Before we can merge this PR, please make sure that all the following items have been
checked off. If any of the checklist items are not applicable, please leave them but
write a little note why.

- [ ] Targeted PR against correct branch (see [CONTRIBUTING.md](https://github.com/cosmos/ibc-go/blob/master/CONTRIBUTING.md#pr-targeting)).
- [ ] Linked to Github issue with discussion and accepted design OR link to spec that describes this work.
- [ ] Code follows the [module structure standards](https://github.com/cosmos/cosmos-sdk/blob/main/docs/docs/building-modules/10-structure.md).
- [ ] Wrote unit and integration [tests](https://github.com/cosmos/ibc-go/blob/master/CONTRIBUTING.md#testing).
- [ ] Updated relevant documentation (`docs/`) or specification (`x/<module>/spec/`).
- [ ] Added relevant `godoc` [comments](https://blog.golang.org/godoc-documenting-go-code).
- [ ] Provide a [commit message](https://github.com/cosmos/ibc-go/blob/main/CONTRIBUTING.md#commit-messages) to be used for the changelog entry in the PR description for review.
- [ ] Re-reviewed `Files changed` in the Github PR explorer.
- [ ] Review `Codecov Report` in the comment section below once CI passes.

### CODE REVIEW

File: modules/core/02-client/migrations/v7/store_test.go [hunk 1]
```diff
@@ -40,23 +36,6 @@ func TestIBCTestSuite(t *testing.T) {
 	suite.Run(t, new(MigrationsV7TestSuite))
 }
 
-// test that MigrateStore returns an error if the codec used doesn't register tendermint types
-func (suite *MigrationsV7TestSuite) TestMigrateStoreTendermint() {
```
Comment (colin-axner) [inline modules/core/02-client/migrations/v7/store_test.go:44]: `GetClientState` uses the client keeper codec which has tendermint client types registered so this test is no longer useful

File: modules/core/02-client/migrations/v7/store.go [hunk 2]
```diff
@@ -82,7 +80,7 @@ func handleSolomachineMigration(ctx sdk.Context, store sdk.KVStore, cdc codec.Bi
 		}
 
 		// update solomachine in store
-		clientStore.Set(host.ClientStateKey(), bz)
+		clientKeeper.SetClientState(ctx, clientID, &updatedClientState)
```
Comment (crodriguezvega) [inline modules/core/02-client/migrations/v7/store.go:83]: If `updatedClientState` is used here, is it still necessary to do `bz, err := clienttypes.MarshalClientState(cdc, &updatedClientState)` above?

Comment (damiannolan) [reply modules/core/02-client/migrations/v7/store.go:83]: I think @crodriguezvega is right here, we can probably remove `bz, err := ...`

**Review by damiannolan — APPROVED** [review]
Nice clean up! Just the one suggestion Carlos made below! Approved anyways, lgtm! 🚀

File: modules/core/02-client/migrations/v7/store.go (hunk 2 above)
Comment (colin-axner) [reply modules/core/02-client/migrations/v7/store.go:83]: Nope, nice catch!

**Review by chatton — APPROVED** [review]

File: modules/core/02-client/migrations/v7/expected_keepers.go [hunk 3]
```diff
@@ -0,0 +1,15 @@
...
+	sdk "github.com/cosmos/cosmos-sdk/types"
+
+	"github.com/cosmos/ibc-go/v6/modules/core/exported"
+)
+
+// ClientKeeper expected IBC client keeper
+type ClientKeeper interface {
+	GetClientState(ctx sdk.Context, clientID string) (exported.ClientState, bool)
+	SetClientState(ctx sdk.Context, clientID string, clientState exported.ClientState)
+	IterateClientStates(ctx sdk.Context, prefix []byte, cb func(string, exported.ClientState) bool)
```
[the remark this answers is not in the document]
Comment (colin-axner) [reply modules/core/02-client/migrations/v7/expected_keepers.go]: nice call! It can be removed since this is a different expected keepers definition

**Pull request merged**